server/src/main/java/org/elasticsearch/index/TimestampBounds.java

```diff
@@ -21,7 +21,9 @@
     TimestampBounds(IndexScopedSettings scopedSettings) {
         startTime = scopedSettings.get(IndexSettings.TIME_SERIES_START_TIME).toEpochMilli();
         endTime = scopedSettings.get(IndexSettings.TIME_SERIES_END_TIME).toEpochMilli();
-        scopedSettings.addSettingsUpdateConsumer(IndexSettings.TIME_SERIES_END_TIME, this::updateEndTime);
+        if (IndexSettings.isTimeSeriesModeEnabled()) {
+            scopedSettings.addSettingsUpdateConsumer(IndexSettings.TIME_SERIES_END_TIME, this::updateEndTime);
+        }
```

Comment:
Not sure why this is necessary. It seems like TimestampBounds instantiation is already guarded by check for the `TIME_SERIES_START_TIME ` in https://github.com/elastic/elasticsearch/blob/3e664d3604ccdd2b35816447ef2f03c658cd88de/server/src/main/java/org/elasticsearch/index/IndexSettings.java#L710
I think @nik9000 had to add this check in #80945 because there was no protection there. 
So

Reply:
@weizijun could you revert this bit? Or explain why it's needed. Like @imotov says I think I think we won't make a `TimestampBounds` at all without the feature flag.

Reply:
But I've been known to be wrong about things.

Reply:
@imotov @nik9000 sorry, I see it just now, I see igor has removed the code